modules/react/common/stories/BasicCompoundComponent.stories.mdx

```diff
@@ -0,0 +1,286 @@
+  contextOverride: usePopupModel.Context,
+})(config => {
+  const model = usePopupModel(config);
+
+  useInitialFocus(model);
+  useReturnFocus(model);
+  useCloseOnOverlayClick(model);
+  useCloseOnEscape(model);
+  useFocusTrap(model);
+  useAssistiveHideSiblings(model);
+  useDisableBodyScroll(model);
+
+  return model;
+});
+```
+
+### TConfig
+
+TConfig gives you the typings that are defined in `defaultConfig` and `requiredConfig` instead of
+having to redfine those types. This is useful when building a model that share similar config and
+you want to merge them while getting the correct typings.
+
+```tsx
+import {createModelHook} from '@workday/canvas-kit-react/common';
+export const useActionBarModel = createModelHook({
+  defaultConfig: {
+    // We define the config and use the typings that come from useMenuModel for when an action bar renders a menu
+    menuConfig: {} as typeof useMenuModel.TConfig,
+  },
+  requiredConfig: useOverflowListModel.requiredConfig,
+})(config => {
+  // define your internal state and events
+});
+```
+
+### getElemProps
+
+This function will separate all `elemProps` or default attributes from an element from the model
+config props. If a prop is both a `config` _and_ an `elemProp`, you can manually apply the prop
+again. `elemProps` is called internally automatically by createContainer. If you use
+`createContainer`, you shouldn't need to use this function.
+
+By default `createModelHook` does this for you and spread the `elemProps` onto the component.
+
+# createContainer
+
+When building a component that has a model, you typically start with a container component.
+`createContainer` is a utility function that hooks up all the pieces for you.
+
+## Usage
+
+- It will wrap your component in a context provider so that `subComponents` have access to the
+  model.
+- It attaches your model hook to your component
+- It runs any `elemPropsHook` hooks defined
+- You can attach `subComponents`.
+- It will also extract the element attributes and define the `ref` type based on the element you
+  give it.
+
+```tsx
+import {createContainer} from '@workday/canvas-kit-react/common';
+interface DisclosureProps {}
+
+//... useDiscloseModel
+
+export const Disclosure = createContainer('div')({
+  displayName: 'Disclosure',
+  modelHook: useDisclosureModel,
+})<DisclosureProps>(({children, ...elemProps}, Element, model) => {
+  return (
+    // spread div attributes including ref
+    <Element {...elemProps}>{children}</Element>
+  );
+});
+```
+
+<ExampleCodeBlock code={CreateContainer} />
+
+# createSubcomponent
+
+Once you've built a container component, adding subcomponent is easy. `createSubcomponent` is
+similar to `createContainer` in that it hooks up a lot of the piece for you. The main difference
+being that is uses the context created by `createContainer`. This allows access to the model created
+at the root level so that you can use any state or events at a child component level.
```

Comment:
```suggestion
Once you've built a container component, adding a subcomponent is easy. `createSubcomponent` is
similar to `createContainer` in that it hooks up many of the pieces for you. The main difference is
it uses the context created by `createContainer`. This allows access to the model created at the
root level so that you can use any state or events at a child component level.
```